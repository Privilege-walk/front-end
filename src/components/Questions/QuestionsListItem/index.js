import React, { useState } from "react";
 import Card from '@mui/material/Card';
 import CardContent from '@mui/material/CardContent';
 import Typography from '@mui/material/Typography';
 import OptionsListItem from "../OptionsListItem";


export default function QuestionsListItem({ description, choices }) {
    const [answerChoices, changeChoices] = useState([...choices].reverse())
    return (
        <Card className="">
            <CardContent>
            <Typography>{description}</Typography>
            </CardContent>
            {
                answerChoices.map( (item) => (
                    <OptionsListItem 
                        key={item.id}
                        type = 'view'
                        description = {item.description}
                        value = {item.value}
                        optionDisabled = {true}
                    />
                ))
            }
        </Card>
    );
}