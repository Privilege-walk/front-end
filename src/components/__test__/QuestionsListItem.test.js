import React from "react";
import { render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

import QuestionsListItem from '../Questions/QuestionsListItem';

test('renders without crashing',  () => {
    let utils;
    act (() => {
        utils = render( <QuestionsListItem description="" choices={[{
            id: 1,
            description: "desc",
            value: 0
        }]} />);
    }); 
    const { asFragment } = utils;
    expect(asFragment()).toMatchSnapshot();
})