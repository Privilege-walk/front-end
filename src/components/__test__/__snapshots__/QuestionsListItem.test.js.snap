// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      />
    </div>
    <div
      style="display: flex; flex-grow: 1; align-items: center;"
    >
      <div
        class="MuiFormControl-root css-104sb8r-MuiFormControl-root"
      >
        <input
          class="form-control"
          data-testid="new-option-desc-undefined"
          disabled=""
          id="new-option-desc-undefined"
          placeholder="Option Description"
          type="text"
          value="desc"
        />
      </div>
      <div
        class="MuiFormControl-root css-3t6boo-MuiFormControl-root"
      >
        <input
          class="form-control"
          data-testid="new-option-points-undefined"
          disabled=""
          id="new-option-points-undefined"
          placeholder="Option Points"
          type="text"
          value="0"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
